src/codegen.cpp

```diff
@@ -7159,10 +7159,11 @@ extern "C" void jl_init_codegen(void)
     initializeAnalysis(Registry);
 #endif
 
-    InitializeNativeTarget();
-    InitializeNativeTargetAsmPrinter();
-    InitializeNativeTargetAsmParser();
-    InitializeNativeTargetDisassembler();
+    InitializeAllTargets();
```

Comment:
> LLVM was built to target only X86 and NVPTX. So, libLLVM-3.9.so didn't contain any function pertaining to other back-ends.
> libjulia.so, built with src/codegen.cpp containing calls to IntializeAll*(), also didn't contain calls to any functions pertaining to backends other than X86 and NVPTX.

LLVM can be built with more backend, as most distros do.

> initialising just the back-ends that the LLVM build targeted

Which can be a lot more than what's needed. Most people don't even need the NVPTX backend to be initialized.

> without using any kind of conditionals.

Which is the worst part of this. Why do you unconditionally initialize things that many people don't need? LLVM isn't particuclarly known for low memory usage and fast compilation (compare to other JIT, not static compilers).